src/Lean/Data/Lsp/LanguageFeatures.lean

```diff
@@ -140,5 +140,79 @@ structure DocumentSymbolResult where
 instance : ToJson DocumentSymbolResult where
   toJson dsr := toJson dsr.syms
 
+inductive SemanticTokenType where
+  | keyword
+  /-
+  | «namespace»
+  | type
+  | «class»
+  | enum
+  | interface
+  | struct
+  | typeParameter
+  | parameter
+  | «variable»
+  | property
+  | enumMember
+  | event
+  | function
+  | method
+  | «macro»
+  | modifier
+  | comment
+  | string
+  | number
+  | regexp
+  | operator
+  -/
+
+def SemanticTokenType.names : Array String :=
+  #["keyword"]
+
+-- must be the correct index in `names`
+def SemanticTokenType.toNat : SemanticTokenType → Nat
+  | keyword => 0
+
+/-
+inductive SemanticTokenModifier where
+  | declaration
+  | definition
+  | readonly
+  | static
+  | deprecated
+  | abstract
+  | async
+  | modification
+  | documentation
+  | defaultLibrary
+-/
+
+structure SemanticTokensLegend where
+  tokenTypes : Array String
+  tokenModifiers : Array String
+  deriving FromJson, ToJson
+
+structure SemanticTokensOptions where
+  legend : SemanticTokensLegend
+  range : Bool
+  full : Bool /- | {
```

Comment:
Not sure I'm reading the spec right but it seems to mean we should have `Either Bool SemanticTokensOptionsDelta` here with the right variant being `structure SemanticTokensOptionsDelta where delta?: Bool` ?

Reply:
This is part of the server capabilities, so no need to support a variant we don't use